sqlitecluster/SQLite.cpp

```diff
@@ -383,39 +510,42 @@ int64_t SQLite::getLastInsertRowID() {
     return sqliteRowID;
 }
 
-// --------------------------------------------------------------------------
 uint64_t SQLite::getCommitCount() {
-    if (_readOnly) {
-        SQResult result;
-        SASSERT(SQuery(_db, "getting commit count", "SELECT MAX(id) FROM journal", result));
-        return !result.empty() ? SToUInt64(result[0][0]) : 0;
-    } else
-        return _commitCount;
+    return _commitCount.load();
+}
+
+uint64_t SQLite::_getCommitCount() {
+    string query = getJournalQuery({"SELECT MAX(id) as maxIDs FROM"}, true);
+    query = "SELECT MAX(maxIDs) FROM (" + query + ") ORDER BY maxIDs desc";
```

Comment:
And shouldn't `desc` be `DESC`?  (Though again, I don't think yo should need it.)